[Owner Review] Added NiDmm missing enum metadata and auto generated them

ShantanuShinde opened the pull request:
### What does this Pull Request accomplish?

The metadata of missing enums and autogenerted them
The name and values of the enums were obtained from the nidmm.h file and they were added to the dictionary in enums.py following the format of other enums. The enums were then referenced in the attributes and functions where they are being used in the attributes.py and functions.py respectively

### Why should this Pull Request be merged?

The missing enums can now be autogenerated

### Testing

The build was successful and the enums got added to the nidmm.proto file

reckenro commented on source/codegen/metadata/nidmm/attributes.py:
```diff
@@ -768,6 +771,7 @@
         },
         'enum': 'AutoZero',
         'lv_property': 'Configuration:Measurement Options:Auto Zero',
+        'enum': 'AutoZero',
```
`'enum': 'AutoZero'` is already specified two lines up.

ShantanuShinde replied:
fixed

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -68,48 +44,30 @@
-                    'description': 'Powerline Cycles'
-                },
-                'name': 'NIDMM_VAL_POWER_LINE_CYCLES',
+               'name': 'NIDMM_VAL_POWER_LINE_CYCLES',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

ShantanuShinde replied:
fixed

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -68,48 +44,30 @@
-                    'description': 'The DMM internally disconnects the input signal following each measurement  and takes a zero reading. It then subtracts the zero reading from the  preceding reading.'
-                },
-                'name': 'NIDMM_VAL_AUTO_ZERO_ON',
+               'name': 'NIDMM_VAL_AUTO_ZERO_ON',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

ShantanuShinde replied:
fixed

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -118,62 +76,86 @@
-                    'description': 'Open and Short Cable Compensation'
-                },
-                'name': 'NIDMM_VAL_CABLE_COMP_OPEN_AND_SHORT',
+               'name': 'NIDMM_VAL_CABLE_COMP_OPEN_AND_SHORT',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -182,107 +164,62 @@
-                    'description': 'AC Current'
-                },
-                'name': 'NIDMM_VAL_AC_CURRENT',
+               'name': 'NIDMM_VAL_AC_CURRENT',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -182,107 +164,62 @@
-                    'description': '2-Wire Resistance'
-                },
-                'name': 'NIDMM_VAL_2_WIRE_RES',
+               'name': 'NIDMM_VAL_2_WIRE_RES',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -182,107 +164,62 @@
-                    'description': 'Frequency'
-                },
-                'name': 'NIDMM_VAL_FREQ',
+               'name': 'NIDMM_VAL_FREQ',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -291,104 +228,94 @@
-                    'description': 'AUX I/O Connector'
-                },
-                'name': 'NIDMM_VAL_EXTERNAL',
+               'name': 'NIDMM_VAL_EXTERNAL',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -291,104 +228,94 @@
-                    'description': 'PXI Trigger Line 5'
-                },
-                'name': 'NIDMM_VAL_PXI_TRIG5',
+               'name': 'NIDMM_VAL_PXI_TRIG5',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -397,87 +324,122 @@
-                    'description': 'Rising Edgs'
-                },
-                'name': 'NIDMM_VAL_POSITIVE',
+               'name': 'NIDMM_VAL_POSITIVE',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -397,87 +324,122 @@
-                    'description': 'IviDmm Mode'
-                },
-                'name': 'NIDMM_VAL_IVIDMM_MODE',
+               'name': 'NIDMM_VAL_IVIDMM_MODE',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro commented on source/codegen/metadata/nidmm/enums.py:
```diff
@@ -397,87 +324,122 @@
-                    'description': 'Waveform acquisition mode'
-                },
-                'name': 'NIDMM_VAL_WAVEFORM_MODE',
+               'name': 'NIDMM_VAL_WAVEFORM_MODE',
```
The spacing changed for 'name'. Make it line up with 'value' beneath.

reckenro reviewed (approved):
Looks good overall. Just some formatting to clean up.

jonlugoatx commented on generated/nidmm/nidmm.proto:
```diff
@@ -229,6 +229,34 @@ enum CableCompensationType {
+
+enum DCBias {
+  option allow_alias = true;
+  D_C_BIAS_UNSPECIFIED = 0;
```
The `D_C_` should be `DC`. Same with `L_C_` below.

ShantanuShinde replied:
this is autogenerated to D_C_. same is the case for the attributes that were already present with DC or LC names. should I change this by editing nidmm.proto manually?

ShantanuShinde replied:
replaced DC with Dc so now DC is generated. done similar for LC, ADCC, and RTD which have similar generation

jonlugoatx reviewed (approved):
1. Please do not remove the testing section from your PR description. That template is there for a reason. If you have done no testing, then say so in that section. In this case, I would expect you to at least say that it builds and runs successfully.
2. You did not comment anywhere on your process for getting these enum values. Did you and copy them from the base enums.hapigen and do hand conversions?
3. How did you arrive at the changes to functions.py? Was that information in the hapigen already but not the nimi-python?

I marked as approved to save time since the changes look good (with a little bit of feedback), but please post responses to these questions. If I have additional feedback based on your answers, then we can address them in a follow-up PR.

ShantanuShinde commented:
> 
> 
>     1. Please do not remove the testing section from your PR description. That template is there for a reason. If you have done no testing, then say so in that section. In this case, I would expect you to at least say that it builds and runs successfully.
> 
>     2. You did not comment anywhere on your process for getting these enum values. Did you and copy them from the base enums.hapigen and do hand conversions?
> 
>     3. How did you arrive at the changes to functions.py? Was that information in the hapigen already but not the nimi-python?
> 
> 
> I marked as approved to save time since the changes look good (with a little bit of feedback), but please post responses to these questions. If I have additional feedback based on your answers, then we can address them in a follow-up PR.

@jonlugoatx 
I added back the testing section in this as well as the other attributes pr and also added the description of how the enums and attributes were added

jonlugoatx reviewed (approved):
Be sure to run the system tests before submitting.
